src/views/Pathways/Courses/Edit/index.js

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect } from 'react';
+import { bindActionCreators } from 'redux';
+import { withRouter } from 'react-router';
+import { compose } from 'recompose';
+import { connect } from 'react-redux';
+import { useSnackbar } from 'notistack';
+import { withTheme } from '@material-ui/core';
+import {find} from 'lodash';
+import { selectors, setCourseToView, addOrEditCourse } from '../../store';
+import { selectors as layoutSelectors, setRightPaneLoading } from '../../../../layouts/TwoColumn/store';
+import { ngFetch } from '../../../../providers/NGFetch';
+import { getCourseEditFormStructure } from '../form/course';
+import FormBuilder from '../../../../components/FormBuilder';
+import Spacer from '../../../../components/Spacer';
+import RightPaneWithTitle from '../../../../components/RightPaneWithTitle';
+import history from '../../../../providers/routing/app-history';
+
+const PathwayCourseEdit = ({ rightPaneLoading, actions, match, theme }) => {
+
+  const { pathwayCourseId, pathwayId } = match.params;
+  const { enqueueSnackbar } = useSnackbar();
+
+  const [coursesAlreadyPresent, setCoursesAlreadyPresent] = React.useState(null);
+
+  const [allCourses, setAllCourses] = React.useState(null);
+  const [course, setCourse] = React.useState(null);
+
+  useEffect(() => {
```

Comment:
The same logic is being used in course addition. Firstly try and see how you can use only one component to do both editing and addition. Don't have two components for it. And all the comments in the add section also apply for the this.